### PR TITLE
Resolves #4069 Now contact types are saved on creation of casa case

### LINKED ISSUE #4069: Bug: Contact Types on casa case do not save/display correctly
### Impacted User Types
 - admins 
 - all casa admins 

### Environment
All, probably. _Seems_ like the environment shouldn't matter to this bug.

### Current Behavior
As a CASA admin, I start by creating a new CASA case. When I only select a few Contact Types...
Screen Shot 2022-10-14 at 2 57 44 PM
...and save the CASA case, ALL checkboxes are checked upon viewing the CASA case again:
Screen Shot 2022-10-14 at 2 59 06 PM

### Expected Behavior
When I create a CASA case and re-visit the edit page, it should have the correct selection of checkboxes checked.

### How to Replicate
1. Log into staging as an admin
2.  Navigate to Cases
3. Click `New Case`
4. Fill in dummy data here but be sure to only select a few of the Contact Type checkboxes
5. Click `Create CASA Case`
6. Click `Edit Case Details`
7. Look at the selected Contact Types--BOOM! 💥 --bug city

### Other notes
My hunch is that something is _wrong_ with the `create` action here and that thing is _right_ with the `edit` action. I believe this because when you edit the record once, the Contact Type checkboxes start persisting. But they should persist on `create`, too, of course!

### QA Login Details:
[Link to QA site](https://casa-qa.herokuapp.com/)  

Login Emails: 
- volunteer1@example.com  view site as a volunteer
- supervisor1@example.com view site as a supervisor
- casa_admin1@example.com view site as an admin

password for all users: 12345678  

### Questions? Join Slack!

We highly recommend that you join us in slack https://rubyforgood.herokuapp.com/ #casa channel to ask questions quickly and hear about office hours (currently Tuesday 6-8pm Pacific), stakeholder news, and upcoming new issues.

### PR DESCRIPTION
**Pull request by ashwinisukale**
Me and @rishijain worked on this PR

### What github issue is this PR for, if any?
Resolves #4069

### What changed, and why?
On creation of case case, we are now assigning the casa contacts too. As it was not being saved before even after the user was selecting the contact types.
On update it was working fine.

### How will this affect user permissions?
- Volunteer permissions: No impact
- Supervisor permissions: No impact
- Admin permissions: No impact

### How is this tested? (please write tests!) 💖💪
Using rspec, lint commands. Also did manual browser testing of the feature.

### CODE REVIEW

**Comment by ashwinisukale** [comment]
I do not have permission to rerun the build, I got a timeout error for this test. Could you please rerun this build
```
1) LearningHoursReport#to_csv when there are learning hours includes all learning hours
     Failure/Error: sleep 0.025 while not_include_fields?(model, result, fields)

     Timeout::Error:
       execution expired
     # ./spec/support/csv_exporter_helper.rb:6:in `sleep'
     # ./spec/support/csv_exporter_helper.rb:6:in `block in wait_for_csv_parse'
     # ./spec/support/csv_exporter_helper.rb:5:in `wait_for_csv_parse'
     # ./spec/models/learning_hours_report_spec.rb:23:in `block (5 levels) in <top (required)>'
     # ./spec/models/learning_hours_report_spec.rb:22:in `each'
     # ./spec/models/learning_hours_report_spec.rb:22:in `each_with_index'
     # ./spec/models/learning_hours_report_spec.rb:22:in `block (4 levels) in <top (required)>'
```

**Review by Thrillberg — COMMENTED** [review]
@ashwinisukale & @rishijain thanks so much for submitting this PR! 😃 I've left a few comments that might improve the code. They should not be blocking, though, since I believe this solution will satisfy the ticket.

@compwron should this PR be approved?

File: spec/controllers/casa_cases_controller_spec.rb [hunk 1]
```diff
@@ -73,5 +73,60 @@
...
+            "date_in_care(1i)" => "2020",
+            "court_dates_attributes" => {"0" => {"date" => "2022/10/31"}},
+            "court_report_status" => "submitted",
+            "casa_case_contact_types_attributes" => [
+              {"contact_type_id" => contact_type.id}
+            ]
+          }
+        }
+
+        it "creates and assigns @case_contact" do
```
[the remark this answers is not in the document]
Comment (ashwinisukale) [reply spec/controllers/casa_cases_controller_spec.rb:97]: Sure, I can do that, thanks for the review

[the remark this answers is not in the document]
Comment (ashwinisukale) [reply spec/controllers/casa_cases_controller_spec.rb:97]: what I am testing here is pass the new case params with case contact details, then it should create the new case, 

can I say like this "It creates new casa case with provided case contacts"

File: app/controllers/casa_cases_controller.rb [hunk 2]
```diff
@@ -48,6 +48,7 @@ def create
     authorize @casa_case
 
     if @casa_case.save
+      @casa_case.update_cleaning_contact_types(casa_case_update_params)
```
[the remark this answers is not in the document]
Comment (ashwinisukale) [reply app/controllers/casa_cases_controller.rb:51]: Yeah, I can check that, thanks!

File: spec/controllers/casa_cases_controller_spec.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (Thrillberg) [reply spec/controllers/casa_cases_controller_spec.rb:97]: Yep, I like that wording better! If we include "with provided case contacts" in this part then maybe we don't need a separate test for the case contacts. Instead we could add an `expect` to verify that the contact type was correctly added?

File: app/controllers/casa_cases_controller.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (ashwinisukale) [reply app/controllers/casa_cases_controller.rb]: Done, addressed this change

File: spec/controllers/casa_cases_controller_spec.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (ashwinisukale) [reply spec/controllers/casa_cases_controller_spec.rb]: Done

**Review by Thrillberg — APPROVED** [review]
Looks great! Thank you! I see this branch is out of date with `main` so once you update your branch this should be ready to go.

**Comment by ashwinisukale** [comment]
Thanks @Thrillberg I have fixed the conflicts.

**Comment by Thrillberg** [comment]
Great @ashwinisukale ! It looks like your change in the code itself has already been made? But the test update is still valuable so I will merge when CI finishes.

**Pull request merged**